buildSrc/src/main/kotlin/smithy-java.publishing-conventions.gradle.kts

```diff
@@ -24,8 +32,7 @@ publishing {
     publications {
         afterEvaluate {
             create<MavenPublication>("mavenJava") {
-                from(components["java"])
-
+                from(extension.customComponent ?: components["java"])
```

Comment:
I don't foresee needing the configurability of the component to be published, so:
```
            if (components.names.contains("shadow")) {
                // Use the shadow component if its exists
                from(components["shadow"])
            } else {
                // Otherwise, use the standard java component
                from(components["java"])
            }
```
should suffice

Reply:
We do. We use the shadow plugin in the CLIs too and we don't want them published to Maven.

Reply:
What I'm asking is if we will ever have more than `"java"` and `"shadow"`